### PR TITLE
BookB-Pset_GeoScienceModelCommon

### PR DESCRIPTION
**Pull request by larswik**
# Pset: Pset_GeoScienceModelCommon

## Description: 

Common properties for geoscience models.

| Property | Datatype | Description |
|----------|----------|-------------|
|ModelPurpose|IfcText|For which Purpose and BIM-Usecase was the model created? E.g. stability assessment, qunatity assessment, tendering etc.|
|CreationDate|IfcDate|Creation date of the model.|
|ReleaseDate|IfcDate|Date of release of the model.|
|UpdateDate|IfcDate|Date of update of the model.|
|VersionNumber|IfcLabel|Version tag of the model regarding its descriptive informations.|
|Purpose|IfcLabel|Purpose for which the model was done. Controlled vocabulary to be consolidated: scientific study, mining, oil reservoir, geotechnics, hydrogeology, simulation etc.|
|Status|IfcLabel|Status of the model. Controlled vocabulary to be consolidated : ended, in progress, in development etc. |
|Author|IfcLabel|Name of the author(s), according to ISO 19115 terminology.<br />>NOTE Maps to PROV:Agent.|
|Organization|IfcLabel|Name of the author(s), according to ISO 19115 terminology.<br />>NOTE Maps to PROV:Agent.|
|AreaOfInterest|IfcIdentifier|Reference to geographical gazeetteer entries (France, Paris basin etc). <br />>NOTE Geonames (https://www.geonames.org/) could be used in case no other is available.|

### CODE REVIEW

File: IFC4x3/Properties/a/Author_1oQcJRV0j0WhD1ng6TcHzF/Documentation.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Name of the author(s), according to ISO 19115 terminology .
```
Comment (SergejMuhic) [inline IFC4x3/Properties/a/Author_1oQcJRV0j0WhD1ng6TcHzF/Documentation.md]:
```suggestion
>NOTE Definition according to ISO 19115: Name of the author(s).
```

File: IFC4x3/Properties/o/Organization_0Ksewl9eXCOBb5HfyICi72/Documentation.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Name of the author(s), according to ISO 19115 terminology.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/o/Organization_0Ksewl9eXCOBb5HfyICi72/Documentation.md]: This description seems wrong.

File: IFC4x3/Properties/p/Purpose_1gm$OqDpv9KQq3uIG1DEsc/Documentation.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Purpose for which the model was done.
+>NOTE Controlled vocabulary to be consolidated: scientific study, mining, oil reservoir, geotechnics, hydrogeology, simulation etc.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/p/Purpose_1gm$OqDpv9KQq3uIG1DEsc/Documentation.md]: Enum?

File: IFC4x3/Properties/s/Status_16vVl01en6dv09d8hlKChk/Documentation.md [hunk 4]
```diff
@@ -0,0 +1,2 @@
+Status of the model. 
+>NOTE Controlled vocabulary to be consolidated : ended, in progress, in development etc.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/s/Status_16vVl01en6dv09d8hlKChk/Documentation.md]: Enum?

File: IFC4x3/Properties/a/Author_1oQcJRV0j0WhD1ng6TcHzF/Documentation.md (hunk 1 above)
Comment (Michel-Rives) [reply IFC4x3/Properties/a/Author_1oQcJRV0j0WhD1ng6TcHzF/Documentation.md]: Proposed change is fine

File: IFC4x3/Properties/o/Organization_0Ksewl9eXCOBb5HfyICi72/Documentation.md (hunk 2 above)
Comment (Michel-Rives) [reply IFC4x3/Properties/o/Organization_0Ksewl9eXCOBb5HfyICi72/Documentation.md]: Please change the description to "Name of the organizations/authors"

File: IFC4x3/Properties/p/Purpose_1gm$OqDpv9KQq3uIG1DEsc/Documentation.md (hunk 3 above)
Comment (Michel-Rives) [reply IFC4x3/Properties/p/Purpose_1gm$OqDpv9KQq3uIG1DEsc/Documentation.md]: Please change description to "Purpose could be, eg: scientific study, mining, oil reservoir, geotechnics, hydrogeology, simulation"
Please rename the property name to "ModelPurpose"

File: IFC4x3/Properties/s/Status_16vVl01en6dv09d8hlKChk/Documentation.md (hunk 4 above)
Comment (Michel-Rives) [reply IFC4x3/Properties/s/Status_16vVl01en6dv09d8hlKChk/Documentation.md]: Please change description to "Status could be, eg: ended, in-progress, in-development"
Please rename the property name to "ModelStatus"

File: IFC4x3/Properties/o/Organization_0Ksewl9eXCOBb5HfyICi72/Documentation.md (hunk 2 above)
Comment (larswik) [inline IFC4x3/Properties/o/Organization_0Ksewl9eXCOBb5HfyICi72/Documentation.md]:
```suggestion
Name of the organizations/authors, according to ISO 19115 terminology.
```

File: IFC4x3/Properties/p/Purpose_1gm$OqDpv9KQq3uIG1DEsc/Documentation.md (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/p/Purpose_1gm$OqDpv9KQq3uIG1DEsc/Documentation.md]: > Please change description to "Purpose could be, eg: scientific study, mining, oil reservoir, geotechnics, hydrogeology, simulation" Please rename the property name to "ModelPurpose"

There is already another property with the name ModelPurpose with a different meaning ("BIM Usecase").

**Comment by larswik** [comment]
From Geo greoup meeting 2023-08-31:
ModelPurpose => ModelUseCase
Purpose: @mbeaufils will look for a definition.

**Comment by mbeaufils** [comment]
Dear @larswik, @j-weil

I agree with @Michel-Rives proposals:
- Delete the current "ModelPurpose",
- Rename the current "Purpose" to "ModelPurpose": Purpose for which the model was done. 
- Rename the current "Status" to "ModelStatus": Status of the model. 
- Authors definition: Name of the author(s).
- Organisation definition: `Name of the organisation(s) involved in producing the model.`

Regarding the codelists:
- ModelPurpose: https://data.geoscience.earth/ncl/ModelPurpose
- ModelStatus: https://data.geoscience.earth/ncl/ModelStatus

**Comment by larswik** [comment]
ToDo @larswik: 
- Renaming ModelPurpose to ModelUseCase according to the meeting with the domain team this morning.
    - I propose to update the description: "Specifies the use case for which the model was created. ModelUseCase could be eg: stability assessment, quantity assessment, tendering etc." - similarly to the other proposals by @Michel-Rives.
- Renaming Purpose to ModelPurpose.
    - Not sure whether the codelists are mandatory or just examples. I suggest to reuse the proposal from Michel and add text about the codelist, such as "Purpose could be, eg: scientific study, mining, oil reservoir, geotechnics, hydrogeology, simulation or a URI from the codelist https://data.geoscience.earth/ncl/ModelPurpose"
- Rename Status to ModelStatus and handle the codelist similarly to ModelPurpose above:
    - "Status could be, eg: ended, in-progress, in-development or a URI from the codelist https://data.geoscience.earth/ncl/ModelStatus"
- Regarding Author/Organisation: Should these actually be list values (since it obviously can be plural)? Should we rename to "Authors" and "Organisations"? Should we also make the Authors description analogue to Organisations, e.g. "Name of the author(s) involved in producing the model". Are these two properties unrelated? ping @mbeaufils.

**Comment by larswik** [comment]
Current definitions:
| Property | Datatype | Description |
|----------|----------|-------------|
|ModelUseCase|IfcText|Specifies the use case for which the model was created. ModelUseCase could be e.g.: stability assessment, quantity assessment, tendering etc.|
|CreationDate|IfcDate|Creation date of the model.|
|ReleaseDate|IfcDate|Date of release of the model.|
|UpdateDate|IfcDate|Date of update of the model.|
|VersionNumber|IfcLabel|Version tag of the model regarding its descriptive informations.|
|ModelPurpose|IfcLabel|Purpose for which the model was done. Purpose could be e.g.: scientific study, mining, oil reservoir, geotechnics, hydrogeology, simulation or a URI from the codelist https://data.geoscience.earth/ncl/ModelPurpose.|
|ModelStatus|IfcLabel|Status of the model. Status could be e.g.: ended, in-progress, in-development or a URI from the codelist https://data.geoscience.earth/ncl/ModelStatus.|
|Authors|List IfcLabel|Name of the author(s) involved in producing the model.|
|Organizations|List IfcLabel |Name of the organization(s) involved in producing the model.|
|AreaOfInterest|IfcIdentifier|Reference to geographical gazeetteer entries (France, Paris basin etc). <br />>NOTE Geonames (https://www.geonames.org/) could be used in case no other is available.|

**Review by pjanck — APPROVED** [review]
lftm

**Review by SergejMuhic — APPROVED** [review]

**Pull request merged**